### PR TITLE
Split adding instruments

### PR DESCRIPTION
**Pull request by bmoneke**
Split "adding_instruments.rst" into several files.

Prepares the way for adding #779 documentation.

To keep the PR small (in scope, not in lines touched), I did not change the content of the files. Links and doctests were changed to continue to work.

This PRalso adds an introductory paragraph according to #824 as an introduction is necessary now. That paragraph is in the new "index" file, before the toc.

I'm open for better implementation ideas.

Let us not change text (except where necessary for the splitting), such that follow up PRs contain textual changes with meaningful diffs. Any textual change in this PR is lost in the splitting diff.

(note in case this PR is dropped: a link to channels had not been set but marked as ToDo)

### CODE REVIEW

File: docs/dev/adding_instruments/index.rst [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+Adding instruments
+##################
+
+You can make a significant contribution to PyMeasure by adding a new instrument to the :code:`pymeasure.instruments` package.
+Even adding an instrument with a few features can help get the ball rolling, since its likely that others are interested in the same instrument.
+
+Before getting started, become familiar with the :doc:`contributing work-flow <../contribute>` for PyMeasure, which steps through the process of adding a new feature (like an instrument) to the development version of the source code.
+
+Pymeasure instruments communicate with the devices via transfer of bytes or ASCII characters encoded as bytes.
+For ease of use, we have :ref:`property creators <properties>` to create easily python properties. Similarly, we have creators to implement easily :ref:`channels <channels>`. Finally, for a smoother implementation process and better maintenance, we have :ref:`tests <tests>`.
```
Comment (bilderbuchi) [inline docs/dev/adding_instruments/index.rst:13]:
```suggestion
For ease of use, we have :ref:`property creators <properties>` to easily create python properties. Similarly, we have creators to easily implement :ref:`channels <channels>`. Finally, for a smoother implementation process and better maintenance, we have :ref:`tests <tests>`.
```

**Comment by bilderbuchi** [comment]
This is perfect, thanks! Thank you for doing the legwork!

**Comment by bilderbuchi** [comment]
Seeing the endless repetition of "This section contains specific documentation on the <manufacturer> nstruments that are implemented....", I'm wondering if there isn't a more elegant way to do this, like generating this sentence automatically for those instrument sections. Maybe reduce the amount of boilerplate files needed when adding an instrument. 
But that's for another issue for sure!

**Comment by bmoneke** [comment]
Thanks to @LongnoseRob, changing all the instrument api files was easy. I used the commands you used for the copyright changes.

**Comment by LongnoseRob** [comment]
> Thanks to @LongnoseRob, changing all the instrument api files was easy. I used the commands you used for the copyright changes.

Great, good to hear!

**Pull request merged**